plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -205,7 +206,9 @@ public IcebergMetadata(
     public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableName tableName)
     {
         IcebergTableName name = IcebergTableName.from(tableName.getTableName());
-        verify(name.getTableType() == DATA, "Wrong table type: " + name.getTableNameWithType());
+        if (name.getTableType() != DATA) {
+            return null;
```

Comment:
Why?

Reply:
When the table redirection towards Hive connector is not enabled,
in case of trying to query on the Iceberg connector
a metadata table of a Hive connector table,
the user will receive a table not found exception.

From `io.trino.tests.product.iceberg.TestIcebergHiveTablesCompatibility#testIcebergSelectFromHiveTable`

```
        assertQueryFailure(() -> onTrino().executeQuery("SELECT * FROM iceberg.default.\"" + tableName + "$data\""))
                .hasMessageMatching("Query failed \\(#\\w+\\):\\Q Not an Iceberg table: default." + tableName);

        assertQueryFailure(() -> onTrino().executeQuery("SELECT * FROM iceberg.default.\"" + tableName + "$files\""))
                .hasMessageMatching("Query failed \\(#\\w+\\):\\Q line 1:15: Table 'iceberg.default." + tableName + "$files' does not exist");
```

Reply:
keeping `verify` in such case seems "OK" too, wdyt?

Reply:
Keeping `verify` would issue for the statement:

```
onTrino().executeQuery("SELECT * FROM iceberg.default.\"" + hiveTableName + "$files\"")
```

error messages like the following:

```
Wrong table type: test_iceberg_select_from_hive_63u5u11q3c70$files
```


I tend to say that the error message 

```
Table 'iceberg.default." + hiveTableName + "$files' does not exist"
```

fits better (not ideal, but better).

Reply:
This requires code comment